File: scripts/generate_gallery_manifest.py
import os
import json

GALLERY_DIR = os.path.join(os.path.dirname(__file__), '..', 'public', 'assets', 'gallery')
MANIFEST_JSON_PATH = os.path.join(GALLERY_DIR, 'gallery.json')
MANIFEST_JS_PATH = os.path.join(GALLERY_DIR, 'manifest.js')

VALID_EXTENSIONS = {'.png', '.jpg', '.jpeg', '.webp', '.gif'}

def generate_manifest():
    manifest = {}
    
    if not os.path.exists(GALLERY_DIR):
        print(f"Gallery directory not found at: {GALLERY_DIR}")
        return

    # Scan subdirectories (spring, summer, autumn, winter, default)
    for folder_name in os.listdir(GALLERY_DIR):
        folder_path = os.path.join(GALLERY_DIR, folder_name)
        
        if os.path.isdir(folder_path):
            images = []
            for filename in os.listdir(folder_path):
                ext = os.path.splitext(filename)[1].lower()
                if ext in VALID_EXTENSIONS:
                    images.append(filename)
            
            # Sort for consistency
            images.sort()
            
            if images:
                manifest[folder_name] = images

    # Write JSON file (optional, but good for reference)
    with open(MANIFEST_JSON_PATH, 'w', encoding='utf-8') as f:
        json.dump(manifest, f, ensure_ascii=False, indent=2)
        
    # Write JS file (for file:// protocol support)
    js_content = f"const GALLERY_MANIFEST = {json.dumps(manifest, ensure_ascii=False, indent=2)};"
    with open(MANIFEST_JS_PATH, 'w', encoding='utf-8') as f:
        f.write(js_content)
    
    print(f"✅ Gallery manifest generated!")
    print(f"   - JSON: {MANIFEST_JSON_PATH}")
    print(f"   - JS:   {MANIFEST_JS_PATH} (Use this for file:// protocol)")
    print(f"   Categories found: {list(manifest.keys())}")
    for cat, imgs in manifest.items():
        print(f"   - {cat}: {len(imgs)} images")

if __name__ == "__main__":
    generate_manifest()

File: scripts/generate-gallery-manifest.js
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const GALLERY_ROOT = path.join(__dirname, "../public/assets/gallery");
const SRC_GENERATED_ROOT = path.join(__dirname, "../src/scripts/generated");
if (!fs.existsSync(SRC_GENERATED_ROOT)) {
    fs.mkdirSync(SRC_GENERATED_ROOT, { recursive: true });
}
const MANIFEST_TS = path.join(SRC_GENERATED_ROOT, "galleryManifest.ts");
const GALLERY_JSON = path.join(GALLERY_ROOT, "gallery.json");

function generateManifest() {
    const manifest = {};
    const seasons = ["default", "spring", "summer", "autumn", "winter"];

    seasons.forEach((season) => {
        const seasonDir = path.join(GALLERY_ROOT, season);
        const files = [];
        if (fs.existsSync(seasonDir)) {
            fs.readdirSync(seasonDir)
                .filter((file) => /\.(png|webp|jpg|jpeg)$/i.test(file))
                .forEach((file) => files.push(file));
            files.sort();
        }
        manifest[season] = files;
    });

    // Write gallery.json (Keep for fallback/reference)
    fs.writeFileSync(GALLERY_JSON, JSON.stringify(manifest, null, 4));
    console.log(`[Manifest Generator] Created: ${GALLERY_JSON}`);

    // Write galleryManifest.ts
    const tsContent = `/**
 * Generated Gallery Manifest
 * Do not edit manually.
 */
export const GALLERY_MANIFEST = ${JSON.stringify(manifest, null, 4)};
`;
    fs.writeFileSync(MANIFEST_TS, tsContent);
    console.log(`[Manifest Generator] Created: ${MANIFEST_TS}`);
}

try {
    generateManifest();
} catch (err) {
    console.error("[Manifest Generator] Error:", err);
    process.exit(1);
}
